src/doc/rustc/src/platform-support/avr-none.md

```diff
@@ -0,0 +1,78 @@
+# `avr-none`
+
+Series of microcontrollers from Atmel: ATmega8, ATmega328p etc.
+
+**Tier: 3**
+
+## Target maintainers
+
+- [Patryk Wychowaniec](https://github.com/Patryk27) <pwychowaniec@pm.me>
+
+## Requirements
+
+This target is only cross-compiled; x86\_64 Linux, x86\_64 macOS and aarch64
+macOS hosts are confirmed to work, but in principle any machine able to run
+rustc and avr-gcc should be good.
+
+Compiling for this target requires `avr-gcc`, because a couple of intrinsics
+(like 32-bit multiplication) rely on [`libgcc`](https://github.com/gcc-mirror/gcc/blob/3269a722b7a03613e9c4e2862bc5088c4a17cc11/libgcc/config/avr/lib1funcs.S)
+and can't be provided through `compiler-builtins` yet. This is a limitation that
+[we hope to lift in the future](https://github.com/rust-lang/compiler-builtins/issues/711).
+
+## Building the target
+
+Rust comes with AVR support enabled, you don't have to rebuild the compiler.
+
+## Building Rust programs
```

Comment:
Just want to mention that I very much appreciate the detailed instructions for building and emulating, I wish all our targets had this.